src/discussions/posts/post-editor/PostEditor.jsx

```diff
@@ -0,0 +1,110 @@
+          </Form.Text>
+
+          <Form.Check inline type="radio" id="question">
+            <Form.Check.Input type="radio" />
+            <Form.Check.Label>
+              <FontAwesomeIcon icon={faQuestion} />
+              <span className="ml-2">{intl.formatMessage(messages.questionType)}</span>
+            </Form.Check.Label>
+          </Form.Check>
+
+          <Form.Check inline type="radio" id="discussion">
+            <Form.Check.Input type="radio" />
+            <Form.Check.Label>
+              <FontAwesomeIcon icon={faComments} />
+              <span className="ml-2">{intl.formatMessage(messages.discussionType)}</span>
+            </Form.Check.Label>
+          </Form.Check>
+        </Form.Group>
+
+        <Form.Group>
+          <Form.Label>
+            {intl.formatMessage(messages.topicArea)}
+          </Form.Label>
+          <Form.Text id="topicAreaInput" muted>
+            {intl.formatMessage(messages.topicAreaDescription)} ({intl.formatMessage(messages.required)})
+          </Form.Text>
+          <Form.Control
+            as="select"
+            defaultValue="General"
+            aria-describedby="topicAreaInput"
+          >
+            {/* TODO: topics has to be filled in another PR */}
+            <option>General</option>
+          </Form.Control>
+        </Form.Group>
+
+        <Form.Group>
+          <Form.Label>
+            {intl.formatMessage(messages.title)}
+          </Form.Label>
+          <Form.Text id="titleInput" muted>
+            {intl.formatMessage(messages.titleDescription)} ({intl.formatMessage(messages.required)})
+          </Form.Text>
+          <Form.Control type="text" aria-describedby="titleInput" />
+        </Form.Group>
+
+        <Form.Group>
+          <Form.Label>
+            {intl.formatMessage(messages.questionText)}
+          </Form.Label>
+          <Form.Control as="textarea" rows="3" />
+        </Form.Group>
+
+        <Form.Group>
+          <Form.Check inline type="checkbox" id="follow">
+            <Form.Check.Input type="checkbox" />
+            <Form.Check.Label>
+              <FontAwesomeIcon icon={faStar} />
+              <span className="ml-2">{intl.formatMessage(messages.followPost)}</span>
+            </Form.Check.Label>
+          </Form.Check>
+
+          <Form.Check inline type="checkbox" id="anonymous">
+            <Form.Check.Input type="checkbox" />
+            <Form.Check.Label>
+              <span className="ml-2">{intl.formatMessage(messages.anonymousPost)}</span>
+            </Form.Check.Label>
+          </Form.Check>
+        </Form.Group>
+
+        <Button className="rounded-lg" variant="primary">
+          {intl.formatMessage(messages.submit)}
+        </Button>
+        <Button className="rounded-lg ml-2" variant="outline-primary" onClick={cancelAdding}>
+          {intl.formatMessage(messages.cancel)}
+        </Button>
+      </Form>
```

Comment:
@0x29a It looks like the form submit button does not have any action. Form submission is part of the PR or you would cover it in other PR?
suggestion: We can use `formik` here. It'll provide JSON object on submit action which will make things easy for dispatch calls otherwise you will need to manually handle all states for fields.

Reply:
@xitij2000 can you please have a look.

Reply:
@awais-ansari This PR was supposed to be only about the UI. The connection to the data will be part of another PR.